Model/TaxCache.php

```diff
@@ -0,0 +1,63 @@
+<?php
+/**
+ * ClassyLlama_AvaTax
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://opensource.org/licenses/osl-3.0.php
+ *
+ * @copyright  Copyright (c) 2018 Avalara, Inc.
+ * @license    http://opensource.org/licenses/osl-3.0.php Open Software License (OSL 3.0)
+ */
+
+namespace ClassyLlama\AvaTax\Model;
+
+use ClassyLlama\AvaTax\Api\TaxCacheInterface;
+
+class TaxCache implements TaxCacheInterface
+{
+    /**
+     * @var \Magento\Customer\Model\Session
+     */
+    protected $customerSession;
+
+    /**
+     * @var \ClassyLlama\AvaTax\Helper\Customer
+     */
+    protected $customerHelper;
+
+    /**
+     * @var \Magento\Framework\App\CacheInterface
+     */
+    protected $cache;
+
+    /**
+     * @param \ClassyLlama\AvaTax\Helper\Customer   $customerHelper
+     * @param \Magento\Customer\Model\Session       $customerSession
+     * @param \Magento\Framework\App\CacheInterface $cache
+     */
+    public function __construct(
+        \ClassyLlama\AvaTax\Helper\Customer $customerHelper,
+        \Magento\Customer\Model\Session $customerSession,
+        \Magento\Framework\App\CacheInterface $cache
+    )
+    {
+        $this->customerSession = $customerSession;
+        $this->customerHelper = $customerHelper;
+        $this->cache = $cache;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function clearCache()
+    {
+        $customerId = $this->customerSession->getCustomer()->getId();
+        $customerCode = $this->customerHelper->getCustomerCode($customerId);
+
+        $this->cache->clean([\ClassyLlama\AvaTax\Helper\Config::AVATAX_CACHE_TAG . "-{$customerCode}"]);
```

Comment:
I realize that the `/rest/V1/avatax/tax/refresh` url will only be called if a customer is logged in, but in case a guest ever does hit this url, I'd like this code to not run. So can you do something like `return false` if `$customerId` is a falsey value?